mindmeld/auto_annotator.py

```diff
@@ -0,0 +1,691 @@
+            time_entities (list): List of allowed time-related entity types.
+
+        Returns:
+            entity (dict): A resolved entity dict or None if the entity isn't resolved.
+        """
+        for candidate in candidates:
+            candidate_entity = SpacyAnnotator._get_time_entity_type(candidate)
+            if (
+                candidate_entity in time_entities
+                and candidate["body"] == entity["body"]
+            ):
+                entity["dim"] = candidate_entity
+                entity["value"] = candidate["value"]
+                return entity
+
+    @staticmethod
+    def _resolve_largest_substring(entity, candidates, entity_types, is_time_related):
+        """ Resolve an entity by the largest substring match given Duckling candidates.
+
+        Args:
+            entity (dict): A dictionary representing an entity.
+            candidates (list): List of dictionary candidates returned by Duckling.parse().
+            entity_types (list): List of entity types to check.
+            is_time_related (bool): Whether the entity is related to time.
+
+        Returns:
+            entity (dict): A resolved entity dict or None if the entity isn't resolved.
+        """
+        largest_candidate = None
+        resolved_entity_type = None
+        for entity_type in entity_types:
+            for candidate in candidates:
+                if is_time_related:
+                    candidate_entity = SpacyAnnotator._get_time_entity_type(candidate)
+                else:
+                    candidate_entity = candidate["entity_type"]
+
+                if (
+                    candidate_entity == entity_type
+                    and candidate["body"] in entity["body"]
+                    and (
+                        largest_candidate is None
+                        or len(candidate["body"]) > len(largest_candidate["body"])
+                    )
+                ):
+                    largest_candidate = candidate
+                    resolved_entity_type = entity_type
+
+        if largest_candidate:
+            entity["body"] = largest_candidate["body"]
+            offset = entity["start"]
+            entity["start"] = offset + largest_candidate["start"]
+            entity["end"] = offset + largest_candidate["end"]
+            entity["value"] = largest_candidate["value"]
+            entity["dim"] = resolved_entity_type
+            return entity
+
+    def _resolve_cardinal(self, entity):
+        if self._resolve_exact_match(entity):
+            return entity
+        candidates = self.duckling.get_candidates_for_text(entity["body"])
+        if self._resolve_largest_substring(
+            entity, candidates, entity_types=["sys_number"], is_time_related=False
+        ):
+            return entity
+
+    def _resolve_money(self, entity, sentence):
+        currency_symbols = ["$", "€", "£", "¥", "₹", "₩", "฿"]
```

Comment:
make this into a class constant

Reply:
Does the resolution here support all currency symbols? If yes, then there is a list of all currency symbols defined in ``constants.py``, you can reuse it.

Reply:
Done